src/pds/ingress/pds-nucleus-datasync-completion.py

```diff
@@ -36,45 +38,36 @@ def lambda_handler(event, context):
 
     logger.info(f"Lambda Request ID: {context.aws_request_id}")
     logger.info(f"Event: {event}")
-    resource = event['resources']
 
-    resource_list = resource[0].split("/")
-    task_id = resource_list[1]
-    exec_id = resource_list[3]
+    json_event = json.dumps(event)
+    transfer_report = json.loads(json_event)
+    content_object = s3.Object('pds-nucleus-datassync-reports', str(transfer_report["s3_key"]))
+    transfer_report_file_content = content_object.get()['Body'].read().decode('utf-8')
+    transfer_report_json_content = json.loads(transfer_report_file_content)
+    verified_file_obj_list = transfer_report_json_content['Verified']
 
-    prefix = f"Detailed-Reports/{task_id}/{exec_id}/{exec_id}.files-verified-"
+    logger.debug(f"verified_file_obj_list: {verified_file_obj_list}")
 
-    datasync_reports_s3_bucket = s3.Bucket(datasync_reports_s3_bucket_name)
+    # Process each file in verified file list
+    for file_obj in verified_file_obj_list:
 
-    list_of_files = []
+        obj_name = file_obj['RelativePath']
+        obj_type = file_obj['SrcMetadata']['Type']
 
-    for transfer_report in datasync_reports_s3_bucket.objects.filter(Prefix=prefix):
+        if obj_type == 'Regular':  # Not a directory
 
-        transfer_report_file_content = transfer_report.get()['Body'].read().decode('utf-8')
-        transfer_report_json_content = json.loads(transfer_report_file_content)
-        verified_file_obj_list = transfer_report_json_content['Verified']
+            if obj_name.endswith('.fz'):
```

Comment:
I don't see an `else` for this `if`; is it okay to ignore files that don't end in `.fz`?

Reply:
@nutjob4life , yes, it is ok to ignore files that don't end in `.fz'. This if condition is just an additional processing step for `.fz` files.

Reply:
Good to know! Thanks @ramesh-maddegoda!